subiquity/common/types.py

```diff
@@ -244,20 +244,29 @@ class Bootloader(enum.Enum):
 
 @attr.s(auto_attribs=True)
 class Partition:
-    size: int
-    number: int
-    annotations: List[str]
+    size: Optional[int] = None
+    number: Optional[int] = None
+    preserve: Optional[bool] = None
+    wipe: Optional[str] = None
+    annotations: Optional[List[str]] = []
+    mount: Optional[str] = None
+    format: Optional[str] = None
+    grub_device: Optional[bool] = None
 
 
 @attr.s(auto_attribs=True)
 class Disk:
     id: str
-    label: str
-    type: str
-    size: int
-    usage_labels: List[str]
     partitions: List[Partition]
-    ok_for_guided: bool
+    label: Optional[str]
+    type: Optional[str]
+    size: int
+    ok_for_guided: Optional[bool]
+    ptable: Optional[str]
+    preserve: Optional[bool] = None
+    path: Optional[str] = None
+    free_for_partitions: Optional[int] = 0
+    usage_labels: Optional[List[str]] = []
```

Comment:
Same comments as before: Optional[int] = 0 looks funny, just int = 0 is fine, use a factory for mutable default.

Reply:
I'm going to change the Disk type declaration back to what it is on main, as there is no scenario where the client sends a json Disk to the server.  Otherwise I agree and would have adopted that suggestion.